### ISSUE TITLE
FileSystemStorageProvider shouldn't allow creating folders with "&" in their name

### ISSUE THREAD

**Issue by HermesSbicego-Laser** [OrchardCMS/Orchard#7572]
The "&" in the name of a MediaLibrary folder creates an error when the Media Url is requested.
repro steps
- Create a folder POI&Events in Media Library
- Upload an image
- request the image URL
you will obtain an error

2 possible solutions:
1. prevent folder creation when the "&" is present
2. encode the "&" as %26 (as done for spaces " " > %20)

**Comment by MatteoPiovanelli-Laser** [comment]
I am working on this, and I found the "&" character in folder names, as is, has another (lesser) issue:

I am checking the files in the "POI&Events" folder at the URL:
https://localhost/Laser.Orchard/NoLaser/Admin/Orchard.MediaLibrary?folderPath=POI&Events
I hit refresh. I find myself at the URL:
https://localhost/Laser.Orchard/NoLaser/Admin/Orchard.MediaLibrary?folderPath=POI

In my case I have no folder there, but this may become an issue if I did.

**Comment by MatteoPiovanelli-Laser** [comment]
Even replacing the "&" with "%26" when generating the MediaUrl there is an error attemtping to access resources there. Something like (the text may not be accurate because my system is in Italian):

```
System.Web.HttpException: Potentially dangerous Request.Path value detected by client (&).

Stack Trace:
[HttpException (0x80004005): Valore potenzialmente pericoloso Request.Path rilevato dal client (&).]
   System.Web.HttpRequest.ValidateInputIfRequiredByConfig() +11999147
   System.Web.PipelineStepManager.ValidateHelper(HttpContext context) +55
```

That leads me to think that the correct solution would be preventing folder names to contain the charater at all.

**Comment by sebastienros** [comment]
I assume this is the case with every char that is not allowed by this filter:
< > * % & : \ ?

**Comment by MatteoPiovanelli-Laser** [comment]
I will verify that

**Comment by MatteoPiovanelli-Laser** [comment]
See updates to #7577

**Comment by BenedekFarkas** [comment]
Fixed in #6792.
